src/mlpack/tests/knn_test.cpp

```diff
@@ -1434,4 +1434,36 @@ BOOST_AUTO_TEST_CASE(MoveOperatorNaiveTest)
   CheckMatrices(distances, distances2);
 }
 
+/**
+ * Check that no garbage value is returned when greedy tree traversal
+ * is performed over kd-tree.
+ */
+BOOST_AUTO_TEST_CASE(GreedyTreeSearch)
+{
+  // Initalize dataset.
+  arma::mat dataset = arma::randu<arma::mat>(3, 100);
+
+  KDTree<EuclideanDistance, NeighborSearchStat<NearestNeighborSort>,
+      arma::mat> tree(dataset);
```

Comment:
To improve the robustness of this test, I'd suggest setting a leaf size of 1 when you build the tree.  It is still possible that you can get leaves in the tree with less than 5 points in them, but it's not guaranteed---so probably better to force a leaf size of 1 to make sure the problematic situation occurs for this test.

Reply:
Sure, I will make that change.